source/speech.py

```diff
@@ -1124,7 +1124,7 @@ def getFormatFieldSpeech(attrs,attrsCache=None,formatConfig=None,unit=None,extra
 	if  formatConfig["reportHeadings"]:
 		headingLevel=attrs.get("heading-level")
 		oldHeadingLevel=attrsCache.get("heading-level") if attrsCache is not None else None
-		if headingLevel and headingLevel!=oldHeadingLevel:
+		if headingLevel and (initialFormat and (reason==controlTypes.REASON_FOCUS or unit in (textInfos.UNIT_LINE,textInfos.UNIT_PARAGRAPH)) or headingLevel!=oldHeadingLevel):

```

Comment:
A comment explaining what this check achieves would be nice; i.e. that as well as speaking if we're entering a new heading, it also always speaks if reading by line or paragraph and the heading is at the start of the unit.

We also should handle "out of heading" for extraDetail, but we could handle that in a separate PR I guess.massive.